x-pack/plugins/fleet/public/components/settings_flyout/hosts_input.tsx

```diff
@@ -158,11 +158,31 @@ export const HostsInput: FunctionComponent<Props> = ({
     [value, onChange]
   );
 
+  const indexedErrors = useMemo(() => {
+    if (!errors) {
+      return [];
+    }
+    return errors.reduce((acc, err) => {
+      if (err.index === undefined) {
+        return acc;
+      }
+
+      if (!acc[err.index]) {
+        acc[err.index] = [];
+      }
+
+      acc[err.index].push(err.message);
+
+      return acc;
+    }, [] as string[][]);
```

Comment:
We could avoid the type assertion here by passing a generic to `reduce`, e.g. `errors.reduce<string[][]>`